modules/atom-ide-ui/pkg/atom-ide-diagnostics-ui/lib/main.js

```diff
@@ -247,34 +272,50 @@ class Activation {
               ),
       );
 
+      Observable.combineLatest(
+        updaters,
+        diagnosticsStream,
+        showTracesStream,
+      ).subscribe(([updater, messages, showTraces]) => {
+        if (updater && showTraces) {
+          updater.fetchDescriptions(messages, false);
+        }
+      });
```

Comment:
1. I could easily add some cleanup code to `this._subscriptions`
2. It was rather easy to add it in here because it already contained all the nice streams to hook into :) but it could be moved somewhere else too.
3. it only fetches them if the check box is checked.

Reply:
At least how it works in `linter-ui-default` and `linter-ui-plus` is that they are only fetched when requested by the user if they are a `Promise`, since there really is no way to know what is coming back from it. 

The idea behind descriptions was that they could be a _long_ form description for what is going on... to the point where the entire help web page could be loaded in there if the provider author really wanted. I don't know of any providers that take advantage of this, but there are ones that provide quite large text in there (`linter-perlcritic` being one of them).

Reply:
sorry, i'm still not sure how to resolve this…if the checkbox is checked, this is going to do some potentially expensive async operation for every diagnostic, every time the diagnostics are updated. that last bit is pretty important because the diagnostics can be updated fairly frequently (e.g. if one is added).

it seems like maybe there's a disconnect about what "description" was intended for and what Nuclide's doing with it. if it truly is designed to be something that's only pulled, then maybe we should have another field for pushable, longer-form summaries? or we could expect summaries to be in a subject + newline + details format and the toggle would affect that…

cc @hansonw and @nmote for some context and advice on this

Reply:
When I left Nuclide, diagnostics were just records with static information. There was no notion of fetching any additional information about diagnostics after the fact. So I don't think I have much to add here, sorry.

Reply:
I somehow like the idea of having 2 descriptions:
* pro: it would make the diagnostics table as simple as it was before
* pro: no async loading of _many_ descriptions because they are not rendered in the table anymore 
* con: there would be 2 descriptions which is kind of hard to explain to a user why the diagnostics datatip description is different to the description in the table (but I'm no UI/UX expert 😬 )

I'm open to do as you wish, I just want the description to be part of the UI _somewhere_ :)

Reply:
Any updates on this? pending PRs are quite tedious...

Reply:
I reverted the code responsible for showing the description in the table. It's only visible in the datatip. How does this sound to you?